### ISSUE TITLE
[REVIEW]: effmass: An effective mass package

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#797]
**Submitting author:** @lucydot (<a href="http://orcid.org/0000-0002-2992-9871">Lucy Whalley</a>)
**Repository:** <a href="https://github.com/lucydot/effmass" target ="_blank">https://github.com/lucydot/effmass</a>
**Version:** v1.0.b1
**Editor:** @katyhuff
**Reviewers:** @ajjackson @bocklund @@mkhorton
**Archive:** Pending

## Status

[![status](http://joss.theoj.org/papers/389754561f0710b756514b8cb9ac0e6a/status.svg)](http://joss.theoj.org/papers/389754561f0710b756514b8cb9ac0e6a)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/389754561f0710b756514b8cb9ac0e6a"><img src="http://joss.theoj.org/papers/389754561f0710b756514b8cb9ac0e6a/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/389754561f0710b756514b8cb9ac0e6a/status.svg)](http://joss.theoj.org/papers/389754561f0710b756514b8cb9ac0e6a)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/lucydot/effmass" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ajjackson, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @katyhuff know.

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/lucydot/effmass">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (v1.0.b1)?
- [ ] **Authorship:** Has the submitting author (@lucydot) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/lucydot/effmass">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (v1.0.b1)?
- [ ] **Authorship:** Has the submitting author (@lucydot) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/lucydot/effmass">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (v1.0.b1)?
- [ ] **Authorship:** Has the submitting author (@lucydot) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks. @ajjackson it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00797/joss.00797/10.21105.joss.00797.pdf)

**Comment by bocklund** [comment]
Here is my first pass:

### General checks

- **Version:** The latest PyPI version is 0.1, behind 0.1.1 and 1.0.b1, but it seems like the the author is working on CI-backed releases. The released version does not work with the current examples on GitHub.

### Functionality

- **Functionality:**  Issue https://github.com/lucydot/effmass/issues/1 was opened. Once closed this can be checked off.

### Documentation

- **A statement of need:** effmass depends on pymatgen through vasppy. @lucydot, can consider comparing effmass to existing software like pymatgen or ASE (what it provides that those do not) in the documentation?  
- **Community guidelines:** There are not very clear guidelines for contributing or a code of conduct (item 1). Support seems to implicitly offered through GitHub, but maybe this is not the author's intention (item 3).

### Other comments

- Since this already depends on pymatgen, is it possible that effmass can support parsing electronic structures from other software?

**Comment by lucydot** [comment]
Hi @bocklund and thank you for taking the time to look over `effmass`

### General checks

- **version**: The latest release contains the changes below and I have uploaded this to PyPI; it should install with `pip install effmass=1.0b3`

### Functionality

- **Functionality**: I have fixed the bug and closed the issue: please see lucydot/effmass#1 for details

### Documentation

- **A statement of need**: Thank you for this idea - I have added a paragraph to the documentation [here](https://effmass.readthedocs.io/en/latest/Features.html).
- **Community guidelines**: I had not thought of including a code of conduct but I think it is a great idea. I have added  [`CODE_OF_CONDUCT.md`](https://github.com/lucydot/effmass/blob/master/CODE_OF_CONDUCT.md) and link to it from the [`README.md`](https://github.com/lucydot/effmass/blob/master/README.md)

### Other comments

- I agree that it would be nice for effmass to support parsing of electronic structures from other software, and it should be relatively easy to implement. I aim to set time aside for a future release, and have suggested it as a possible contribution in the [`README.md`](https://github.com/lucydot/effmass/blob/master/README.md).

**Comment by bocklund** [comment]
@lucydot, thanks for your response. Looks good to me.

cc @katyhuff

**Comment by katyhuff** [comment]
Excellent! Thank you for your review @bocklund . 

@ajjackson are you still interested in completing a review of this work?

@mkhorton -- it looks like you've begun your review. How is the remainder of the review process going?

**Comment by ajjackson** [comment]
Yes, I have been looking at it today!

As Lucy mentioned in the Pre-review thread, we were members of the same research group a little over two years ago but we never published together. I was not aware of the existence of this code until asked to review it.

I will check off what I can now and then try out the code on some fresh data.

**Comment by ajjackson** [comment]
As a developer of Sumo (which is also going through JOSS review), I am comfortable with the amount of overlap between effmass and Sumo. Effmass is more flexible for effective mass fitting and concentrates on providing a toolkit for analysis. Sumo is a suite of command-line tools; it is more opinionated and "fire-and-forget", with a less Python-adept target audience.

**Comment by ajjackson** [comment]
In general **effmass** seems to be packaged and documented very nicely. I need to take a bit more time to play with the actual functionality. Here are some comments in the meantime:

## General checks
- **version**: the PyPI/beta/source versioning was a touch confusing but as far as I can see the packaging all works as it should and I was able to get the 1.01 beta versions easily enough. I think this is ok.

## Documentation
- **Statement of need**: The summary of unique features is helpful, a very useful thing to have when people are looking for tools. I would like to see a more explicit statement somewhere of the target audience (i.e. theoretical solid-state physics/chemistry community). The review guidelines imply that this should be in both the paper and other docs.
- **Installation instructions**: 
  - The Python package management works well and I don't think a separate requirements list is essential. I do think it is worth mentioning that Numpy/Matplotlib are used as these dependencies are a) quite large b) sometimes best installed with a different package manager.
  - Numpy is listed as a dependency in setup.py and requirements.txt, but SciPy is not. Scipy is imported in `analysis.py` so seems to be a required module. It _is_ one of Pymatgen's many dependencies so the installers work, but this seems inconsistent when other packages installed by Pymatgen are listed explicity.
  - requirements.txt is missing Pymatgen. I assume this is related to making readthedocs work, as we ran into exactly this issue on Sumo. Probably the best solution is to name it requirements_rtd or something so that people don't try to use it with virtualenv.
- **Community guidelines:** The guidelines for contributors are good. The guidelines for bugs/support are minimal but do the job.

@bocklund and @mkhorton have both noted the Pymatgen dependency as a basis for further development. Poking around though, I'm not sure that effmass actually uses Pymatgen at all? Perhaps I've missed something, but as far as I can see it's just a side-effect of installing Vasppy?

**Comment by mkhorton** [comment]
@ajjackson yeah, it's a little weird -- I'm not actually familiar with vasppy, but it looks like pymatgen is a dependency of vasppy but it doesn't actually use pymatgen much. In particular, vasppy itself doesn't seem to build on pymatgen's VASP classes, but rather has its own VASP classes, which means building in pymatgen support to effmass would indeed be a bit trickier. (Hello by the way, it's a small world -- I didn't realise you were the developer of sumo!)

**Comment by lucydot** [comment]
Thank you @ajjackson and @mkhorton for your suggestions. 

### Overlap with other packages
I've added a `Related packages` section to the [paper.md](https://github.com/lucydot/effmass/blob/master/paper.md). This has also been added as a [new page in RTD](https://effmass.readthedocs.io/en/latest/Related%20packages.html), and a link to this page has been added to the [README](https://github.com/lucydot/effmass/blob/master/README.md).

### Target audience
I have explicitly stated the target audience as part of the `Related packages` documentation (see above).

### Installation instructions
I have included explicit mention of the scipy, numpy and matplotlib packages as suggested.

### Dependencies: scipy
Thank-you for pointing out I had missed listing scipy as a dependency in in requirements.txt and setup.py - this was an oversight and it is now added. 

### Dependancies: pymatgen
I haven't included pymatgen as effmass does not make use of this (it is only installed as it is a dependency of vasppy).  To enable easy extension to other electronic structure code outputs, there is definitely an argument for losing the vasppy dependency and using pymatgen instead, though this would involve re-writing a significant part of the code.

**Comment by katyhuff** [comment]
Thank you @lucydot . It looks like you've addressed the reviewer comments. Let's see if they can recommend publication.

@mkhorton & @ajjackson : Thank you for your reviews. Do you have any remaining comments? Are you ready to make a final recommendation for acceptance based on this response, or would you like more time to explore the package?

**Comment by mkhorton** [comment]
I've updated my checklist. Ideally, I'd like some further time to play with this to check off the remaining items, but I'm happy to defer to the other referees if they're happy to accept now -- it certainly seems like a very nicely structured package and will be a valuable addition to the community.

**Comment by katyhuff** [comment]
Thanks @mkhorton . Feel free to explore a bit more. I'll check back in next week.

**Comment by ajjackson** [comment]
I have worked through the Jupyter tutorial and was able to follow it and reproduce the plots without any trouble. It gives an applied overview of features; the comparison with DOS filling is a nice touch.

- [Issue raised](https://github.com/lucydot/effmass/issues/4) over broken link to data
- I can reproduce step [5] fine but it's not clear how it should be interpreted. The docstring says _"Plots bandstructure with the DFT-calculated points for each Segment instance overlaid."_ but if I had to guess what the vector labels are I'd say they are directions? It would be really helpful if these labels also included the segment indices.
- I have raised a feature request [here](https://github.com/lucydot/effmass/issues/3) which I think would improve usability. This should not be considered a condition of acceptance, it's a "nice to have".

I'd like to throw some of my own data into the mix before signing off

**Comment by lucydot** [comment]
Thanks for the feedback @ajjackson:

- I have fixed the link and closed the issue.
- The labels now include the segments argument index. See [the notebook](https://github.com/lucydot/effmass/blob/master/paper/notebook.ipynb) for the example output - have I interpreted your suggestion correctly? The labels tended to overlap, so I have used the adjustText package to avoid this, and I think the output is much easier to read now. The function docstring and notebook have been updated so that it is clear what the labels correspond to.
- I have implemented the feature request: https://github.com/lucydot/effmass/issues/3 - thanks for the suggestion, it echos a similar (verbal) suggestion from @Guaxixi .

**Comment by ajjackson** [comment]
Updates:

**Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
In general effmass is following best practice in the use of Python docstrings and automatically-generated documentation. I have checked this off as satisfactory, with the following observations:
- I found a minor error in a docstring, raised as an issue here: https://github.com/lucydot/effmass/issues/6 
- It would be nice to have a bit more description in the API documentation re: what the modules are for. My own code documentation is also sparse in this respect... something to thing about for the future!

** References** It's difficult to check the references since the paper was updated as the PDF manuscript hasn't been rebuilt. I don't see a formal academic citation for the Seebeck effective mass paper (is that deliberate to indicate that it isn't used in this package?), or for pymatgen (https://doi.org/10.1016/j.commatsci.2012.10.028).

**Functionality** My attempt to test with independent data was foiled by some issues with partial occupancies; I will have another go before signing off if that's ok. https://github.com/lucydot/effmass/issues/5

**Comment by katyhuff** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by katyhuff** [comment]
@ajjackson please see built pdf above (to see what whedon can do, you can execute `@whedon commands` at any time)

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00797/joss.00797/10.21105.joss.00797.pdf)

**Comment by katyhuff** [comment]
@ajjackson  regarding your concern about api docs, the readme for the repo links to these API docs which seem quite sufficient to me : https://effmass.readthedocs.io/en/latest/

**Comment by ajjackson** [comment]
I agree they are sufficient, it's not a big deal  :-) I was referring to pages such as https://effmass.readthedocs.io/en/latest/API%20documentation.html which currently contains links without descriptions. This is very common in automatically generated documentation.

**Comment by katyhuff** [comment]
Ah, I misunderstood. Your comment initially made me think you couldn't find your way to the API pages.

Yes, I agree this is the default sphinx behavior. More might be better on that page, perhaps. But in this case, I think the docs are strong and it's not necessary if the reader has the patience to click through the links.

**Comment by mkhorton** [comment]
@lucydot running `python setup.py test` fails with:

```
  File ".../effmass/tests/test_inputs.py", line 10, in <module>
    (pytest.lazy_fixture('MAPI_settings_object'), 0.025, 0.25),
AttributeError: module 'pytest' has no attribute 'lazy_fixture'
```

Your tests are working correctly with `python -m pytest` as recommended by your README, so this is not a bug, but I thought I'd bring this to your attention since it might trip people up.

**Comment by ajjackson** [comment]
I've raised an issue [here](https://github.com/lucydot/effmass/issues/7) about spin-polarised calculations, happy to tick the last box once that's resolved!

I intend to use this package.  It provides convenient comparison of different fitting methods, which is a good sanity check, and implements a superior weighted method for effective-mass determination. The Kane model implementation provides a useful way of quantifying non-parabolicity. The packaging follows best practices, and decent API documentation is accompanied by a useful tutorial.

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00797/joss.00797/10.21105.joss.00797.pdf)

**Comment by lucydot** [comment]
Thanks for the feedback @mkhorton and @ajjackson, it is really useful. @ajjackson I am chuffed you are planning to make use of it :smiley:

@mkhorton thanks for pointing out the possible testing pitfall,  I've raised it as an issue on the repo.

**Functionality**:

- I have looked at the issues raised @ajjackson ([here](https://github.com/lucydot/effmass/issues/5) and [here](https://github.com/lucydot/effmass/issues/7)), let me know whether these solve the problem and if so I will close them.

**Documentation**:

- I have fixed the issue regarding out of date docstring [here](https://github.com/lucydot/effmass/issues/6).
- I have added some module level docstrings to the API documentation [here](https://effmass.readthedocs.io/en/latest/API%20documentation.html).

**References**

- I have included academic references for pymatgen. I will add a reference for sumo once it has a Zenodo / JOSS DOI.
- I haven't included the Seebeck reference because, as you say, it isn't implemented in effmass. I do include a link to a paper which contains a definition and further references.

**Comment by katyhuff** [comment]
This looks like a very comprehensive response @lucydot . I think that these changes are sufficient and hopefully @ajjackson will confirm. Assuming this is confirmed, I think this is ready to accept. I know you intend to include a reference to the sumo JOSS paper, but I don't think the review process for sumo should delay the review process for this work. If a reference to the sumo repository can be added, perhaps that is a good middle ground.

In any case, as soon as we get confirmation from @ajjackson that these issues satisfy the concerns, this will be ready to accept. At that point could you please make an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive? I can then move forward with accepting the submission.

**Comment by katyhuff** [comment]
It looks like there are still two open issues that you're communicating in ([here](https://github.com/lucydot/effmass/issues/5) and [here](https://github.com/lucydot/effmass/issues/7]). Perhaps one of you (@lucydot @ajjackson) when we're ready to move forward with the submission.

**Comment by lucydot** [comment]
Yes, I'd like to sort out the discrepancy with the sumo package in [issue #7](https://github.com/lucydot/effmass/issues/7)  - it looks like effmass is not handling spin-polarised data properly.

**Comment by ajjackson** [comment]
Yes, [Issue 5 (partial occupancy)](https://github.com/lucydot/effmass/issues/5) is fine now. Just waiting on [issue 7](https://github.com/lucydot/effmass/issues/7) which affects the calculated values.

**Comment by lucydot** [comment]
Latest commit to master fixes [issue 7](https://github.com/lucydot/effmass/issues/7)- thank-you for finding this bug @ajjackson, it was an important one!

**Comment by lucydot** [comment]
Issue 7 is not quite closed (but very close!), when it is I will archive and update this thread.

**Comment by katyhuff** [comment]
Ah, I misunderstood the previous comment.
Thanks.

**Comment by ajjackson** [comment]
Issue 7 looks good now, I have ticked off "functionality".

**Comment by katyhuff** [comment]
Excellent. @lucydot when you make archive of the reviewed software in Zenodo/figshare/other service, please just update this thread with the DOI of the archive and we'll move forward with acceptance! (You may want to be sure to double check the paper, all minor details, etc. before creating the archive).

**Comment by katyhuff** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00797/joss.00797/10.21105.joss.00797.pdf)

**Comment by lucydot** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00797/joss.00797/10.21105.joss.00797.pdf)

**Comment by lucydot** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #797 with the following error: 

   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    12    0    12    0     0    238      0 --:--:-- --:--:-- --:--:--   240
Error reading bibliography ./paper.bib (line 61, column 1):
unexpected "a"
expecting space, ",", white space or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by lucydot** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00797/joss.00797/10.21105.joss.00797.pdf)

**Comment by lucydot** [comment]
Hi @katyhuff  - here is the DOI: 10.5281/zenodo.1338919
Thank-you for guiding the review process, `effmass` gained much from it.

**Comment by katyhuff** [comment]
@whedon set 10.5281/zenodo.1338919 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.1338919" target="_blank">10.5281/zenodo.1338919</a> is the archive.

**Comment by katyhuff** [comment]
@arfon This paper is ready for acceptance!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.00797/status.svg)](https://doi.org/10.21105/joss.00797)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.00797">
  <img src="http://joss.theoj.org/papers/10.21105/joss.00797/status.svg" alt="DOI badge" >
</a>
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.00797/status.svg)](https://doi.org/10.21105/joss.00797)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=

**Comment by katyhuff** [comment]
Thanks @ajjackson @bocklund @mkhorton! And congratulations @lucydot !
